app/assets/javascripts/miq_application.js

```diff
@@ -320,7 +320,10 @@ function miqCheckForChanges() {
     }
   }
   // use default browser reaction for onclick
-  return true;
+  if(obj && (href = obj.getAttribute("href")))
```

Comment:
@himdel It looks like we simply cannot use ```$stateChangeStart``` and ```$stateProvider.state("otherwise")``` in this scenario due to the state/template unloading issue described above.

```$locationChangeSuccess``` however, looks promising and seems to be working very well.
```javascript
$rootScope.$on('$locationChangeSuccess', function(event, newURL, oldURL) {
    if (newURL != oldURL) {
      window.location.href = newURL;
    }
  });
```
I think it is by far the cleanest solution to this issue. What say you?